docs/en/overview/Glossary.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: global
+title: Glossary
+group: Overview
+priority: 1
+---
+
+<!-- * Table of Contents
+{:toc} -->
+
+Glossary of Alluxio terminology
```

Comment:
pages don't usually start with a line of normal text? i think i'd just remove this since it'll be titled as `Glossary` anyways